### PR TITLE
Task 3/logging subsystem

### PR DESCRIPTION
**Pull request by MrDmitry**

### CODE REVIEW

File: MuleUtilities/asset/source/SimpleStorage.cpp [hunk 1]
```diff
@@ -22,25 +22,25 @@ Handler SimpleStorage::Get(const KeyType& key)
 
     if (m_entries.cend() != it)
     {
-        LOG_DEBUG("Requested asset \"%s\" -> exists", key.c_str());
+        g_loggers[Log::mule_storage]->Debug("Requested asset \"{}\" -> exists", key.c_str());
```
Comment (ilia-glushchenko) [inline MuleUtilities/asset/source/SimpleStorage.cpp]: For me, it looks way too long, maybe add some alias or typedef or something?

File: MuleUtilities/source/Logger.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#include <mule/Logger.hpp>
+
+#include <spdlog/sinks/null_sink.h>
+
+namespace mule
+{
+
+LoggerPtr g_logger = std::make_shared<Logger>(
+    "mule"
+    ,
```
Comment (ilia-glushchenko) [inline MuleUtilities/source/Logger.cpp]: Move it to the previous line?

**Review by ilia-glushchenko — CHANGES_REQUESTED** [review]

File: MuleUtilities/asset/source/SimpleStorage.cpp (hunk 1 above)
Comment (MrDmitry) [reply MuleUtilities/asset/source/SimpleStorage.cpp]: Added macros

File: MuleUtilities/source/Logger.cpp (hunk 2 above)
Comment (MrDmitry) [reply MuleUtilities/source/Logger.cpp]: Removed this file altogether

**Comment by ilia-glushchenko** [comment]
Is there any particular reason why Fix clang warning commit does not have issue reference?

**Comment by krupitskas** [comment]
Squash and merge, anyway we havent here any pullaprove setup

**Comment by MrDmitry** [comment]
`Is there any particular reason why Fix clang warning commit does not have issue reference?`

Generally non-functional commits that are subject to be squashed will create garbage unneeded references in the issue itself. It's better to have functional commits there than random ones.

**Comment by ilia-glushchenko** [comment]
Okay, as long as they don't go to master, I'm good with it.

**Review by ilia-glushchenko — APPROVED** [review]

**Pull request merged**